cloudferrylib/os/compute/nova_compute.py

```diff
@@ -504,10 +505,14 @@ def get_instances_list(self, detailed=True, search_opts=None, marker=None,
                 detailed=detailed, search_opts=search_opts, marker=marker,
                 limit=limit)
         else:
-            if type(ids) is list:
-                instances = [self.nova_client.servers.get(i) for i in ids]
-            else:
-                instances = [self.nova_client.servers.get(ids)]
+            ids = ids if type(ids) is list else [ids]
+            instances = []
+            for i in ids:
+                try:
+                    instances.append(self.nova_client.servers.get(i))
+                except nova_exc.NotFound:
+                    raise cf_exc.NotFound("No server with ID of '%s' exists."
```

Comment:
I guess we need to skip this error with printing in the Warning log. At least, if we want to have it here. Otherwise, we need to have it on the filter level. Or we can implement universal checking action for such purposes, cause currently we also have images filtering too, and in future there will be more different objects (f.e. we plan add volumes filtering).